File: DB/src/main/java/adapter/converter/itemProto/ConverterItem.java
package adapter.converter.itemProto;

import Domain.Dto.ItemCreateDto;
import Domain.Model.Item;
import Domain.Model.ItemType;
import Domain.Model.Shelf;
import Domain.Model.User;
import GRPC.proto.File.*;
import adapter.converter.ShelfProto.ConverterShelf;
import adapter.converter.itemTypeProto.ConverterItemType;
import adapter.converter.userProto.ConverterUser;
import database.DaoInterface.IItemDao;

public class ConverterItem
{
  public static ItemProto item_To_ItemProto(Item item)
  {
    return ItemProto.newBuilder()
        .setType(ConverterItemType.itemType_To_ItemTypeProto(item.getType()))
        .setUniqueID(item.getUid())
        .setOwner(ConverterUser.User_To_UserProto(item.getOwner()))
        .setShelf(ConverterShelf.shelf_to_ShelfProto(item.getShelf()))
        .build();
  }



  public static Item itemProto_to_Item(ItemProto itemProto)
  {
    ItemType itemType = ConverterItemType.itemTypeProto_To_ItemType(itemProto.getType());
    User user = ConverterUser.UserProto_To_User(itemProto.getOwner());
    Shelf shelf = ConverterShelf.shelfProto_To_shelf(itemProto.getShelf());


    return new Item(
        itemType,
        itemProto.getUniqueID(),
        user,
        shelf
    );

  }

  public static ItemCreateDto itemProtoCretion_to_Item(ItemCreation itemProto)
  {


    return new ItemCreateDto(
            itemProto.getItemTypeID(),
            itemProto.getOwnerID(),
            itemProto.getShelfID()
    );

  }
}

File: Logic/AdapterToGRPC/Converter/ConverterItem.cs
using System.Collections;
using GRPC.Proto;
using Shared.DTOs;
using Shared.Model;

namespace Logic.AdapterToGRPC;

public class ConverterItem
{
    public static Shared.Model.Item ItemProtoToItem(ItemProto itemProto)
    {
        List<Shared.Model.Item> items = new List<Shared.Model.Item>();
        
        User user = ConverterUser.UserProtoToUser(itemProto.Owner);
            
        itemType itemType = ConverterItemType.ItemTypeProtoToItemType(itemProto.Type);

        Shared.Model.Shelf shelf = new Shared.Model.Shelf();
        shelf.DimX = itemProto.Shelf.DimX;
        shelf.DimY = itemProto.Shelf.Dimz;
        shelf.DimZ = itemProto.Shelf.Dimz;
        shelf.RowNo = itemProto.Shelf.RowNo;
        shelf.ShelfNo = itemProto.Shelf.ShelfNo;


        Shared.Model.Item result = new Shared.Model.Item(itemType, itemProto.UniqueID, user, shelf);

        

        foreach (ItemProto item in itemProto.Shelf.ItemsOnShelf)
        {
            User _user = ConverterUser.UserProtoToUser(item.Owner);
            
            itemType _itemType = ConverterItemType.ItemTypeProtoToItemType(item.Type);  
            
            items.Add(new Shared.Model.Item(_itemType,itemProto.UniqueID,_user,shelf));
        }

        shelf.ItemsOnShelf = items;

        return result;
    }

    public static ItemCreation ItemCreationDtoToItemCreation(ItemCreationDto item)
    {
        return new GRPC.Proto.ItemCreation
            { ItemTypeID = item.ItemTypeId, ShelfID = (int)item.shelfId, OwnerID = item.OwnerId };
    }
}